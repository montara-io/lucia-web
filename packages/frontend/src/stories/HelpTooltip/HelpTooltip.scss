@import '../../styles/colors';
@import '../../styles/global';

.m-help-tooltip {
  margin-top: -2px;
  margin-right: 23px;
  right: 0;
  transform: translate(50%, -100%);
  width: max-content;
  background-color: $secondary;
  font-size: 0.65rem;
  position: absolute;
  font-family: inherit;
  font-style: inherit;
  font-weight: 500;
  letter-spacing: normal;
  text-align: start;
  text-shadow: none;
  text-transform: none;
  white-space: normal;
  word-break: normal;
  word-spacing: normal;
  word-wrap: normal;
  display: inline-block;
  z-index: 111;

  &.left {
    margin-top: 14px;
    transform: initial;
    right: 28px;
  }

  .m-help-tooltip-inner {
    max-width: 117px;
    padding: $spacing-unit-8 / 2 + 1 $spacing-unit-8;
    color: $white;
    text-align: center;
  }
}
