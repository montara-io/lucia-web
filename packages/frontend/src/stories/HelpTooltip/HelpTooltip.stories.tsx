import * as React from 'react'
import HelpTooltip from './HelpTooltip'

export default {
  title: 'Components/HelpTooltip',
  component: HelpTooltip,
}

export const Basic = () => (
  <div style={{ position: 'absolute', marginTop: '15px' }}>
    <HelpTooltip>Some text here</HelpTooltip>
  </div>
)

Basic.args = {}

Basic.parameters = {
  jest: ['HelpTooltip.test.tsx'],
}

export const Left = () => (
  <div style={{ position: 'absolute', marginTop: '15px' }}>
    <HelpTooltip helpTooltipPosition={'top'}>Some text here</HelpTooltip>
  </div>
)
