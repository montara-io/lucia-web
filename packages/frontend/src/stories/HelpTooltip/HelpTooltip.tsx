import classNames from 'classnames'
import React, { CSSProperties } from 'react'
import './HelpTooltip.scss'

export interface HelpTooltipProps {
  style?: CSSProperties
  className?: string
  helpTooltipPosition?: 'top' | 'left' | undefined
  children?: any
}

const HelpTooltip: React.FunctionComponent<HelpTooltipProps> = ({
  children,
  style,
  className,
  helpTooltipPosition = 'top',
}) => {
  return (
    <div
      className={classNames(
        'm-help-tooltip',
        className,
        helpTooltipPosition === 'left' ? 'left' : '',
      )}
      data-testid="m-help-tooltip"
      {...(style ? { style } : {})}
    >
      <div className="m-help-tooltip-inner">{children}</div>
    </div>
  )
}

export default HelpTooltip
