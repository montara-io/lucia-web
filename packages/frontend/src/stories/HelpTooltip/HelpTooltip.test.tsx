import React from 'react'
import HelpTooltip, { HelpTooltipProps } from './HelpTooltip'

describe('Test HelpTooltip Component', () => {
  let props: HelpTooltipProps

  beforeEach(() => {
    props = {}
  })

  it('Snapshot renders correctly', () => {
    expect(<HelpTooltip {...props}>Text</HelpTooltip>).toMatchSnapshot()
  })
})
