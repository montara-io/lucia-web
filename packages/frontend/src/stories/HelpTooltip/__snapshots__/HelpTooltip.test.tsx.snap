// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HelpTooltip Component Snapshot renders correctly 1`] = `
<HelpTooltip>
  Text
</HelpTooltip>
`;
